### ISSUE TITLE
Boulton (V17.0) repeating callbacks for older blocks

### ISSUE THREAD

**Issue by maupin** [nanocurrency/nano-node#1489]
After upgrading to Boulton (on Debian 9) I notice that the RPC callback is sending out repeated notices of the same, older blocks.  My listener logs each callbacked block hash with a Unix timestamp, like this:

```
1545211772512: 3A3FC8FD81627B974616125B926095D8911545637F750F84C1579F03C456C72E
1545211773372: 16B82DE1ADB5F4A5C00C65CE3C51A2A9E5B3CDD9FABD9943938E0F62E5670C6C
1545211782698: 6B120C9AF281D56F0C395BE54103E651BD4108350E5750D75EBB2E97ED87BF87
1545211782751: DC4702123DB34962C5B135770D712313662DFAEC8FBCE250551390D3E9589350
```
... etc

I notice that, for example, block hash 3A3FC8FD81627B974616125B926095D8911545637F750F84C1579F03C456C72E (originally seen in August!) and others appear multiple times in the logs.  For example,

```
1545211772512: 3A3FC8FD81627B974616125B926095D8911545637F750F84C1579F03C456C72E
-skip ahead a bit-
1545211850967: 3A3FC8FD81627B974616125B926095D8911545637F750F84C1579F03C456C72E
-skip ahead a bit-
1545213648643: 3A3FC8FD81627B974616125B926095D8911545637F750F84C1579F03C456C72E
-skip ahead a bit-
1545213720942: 3A3FC8FD81627B974616125B926095D8911545637F750F84C1579F03C456C72E
-skip ahead a bit-
1545214044117: 3A3FC8FD81627B974616125B926095D8911545637F750F84C1579F03C456C72E
```
... etc

**Comment by SergiySW** [comment]
Should be fixed after https://github.com/nanocurrency/raiblocks/pull/1303 merging

**Comment by zhyatt** [comment]
Fixed with #1770 and #1801
